cadcutils/cadcutils/net/auth.py

```diff
@@ -100,33 +178,37 @@ def get_cert(cert_server=None,
     cert_server = cert_server is None and CERT_SERVER or cert_server
     cert_endpoint = cert_endpoint is None and CERT_ENDPOINT or cert_endpoint
 
-    username, passwd = get_user_password(cert_server)
+    username = user
+    if username is None:
+        sys.stdout.write("CADC user ID: ")
+        username = sys.stdin.readline().strip('\n')
+
+    username, passwd = get_user_password(cert_server, username=username)
 
     url = "http://{0}/{1}".format(cert_server, cert_endpoint)
     resp = requests.get(url, params=kwargs, auth=(username, passwd))
     resp.raise_for_status()
```

Comment:
check_status is mapping the requests.HTTPError into our own generic exceptions. Because of that, a user does not have to be aware of all the possible HTTP error codes. In addition, the requests library is not exposed through the API to the user of the BaseWsClient class.

As for get_cert, it should use the BaseWsClient now that it's available. I'm not going to do it now because it is simple to implement it when the registry functionality is available so I created a task for this in that story.